### PR TITLE
add kind logs

### PR DESCRIPTION
**Pull request by kannon92**
Grab logs of kind cluster and follow similar pattern to Kueue (https://github.com/kubernetes-sigs/kueue/blob/cc58315b5c572fd872e354bfafaea5582c04a60c/hack/e2e-test.sh#L27-L40).

### CODE REVIEW

**Comment by ahg-g** [comment]
I don't see any meaningful logs from kube-scheduler for example: https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_jobset/214/pull-jobset-test-e2e-main-1-24/1674410094915227648/artifacts/kube-scheduler.log

File: hack/e2e-test.sh [hunk 1]
```diff
@@ -4,15 +4,24 @@ export KIND=$PWD/bin/kind
...
-    then 
-        $KIND delete cluster --name $KIND_CLUSTER_NAME
+    then
+        if [ ! -d "$ARTIFACTS" ]; then
+            mkdir -p "$ARTIFACTS"
+        fi
+        kubectl logs -n kube-system kube-scheduler-kind-control-plane > $ARTIFACTS/kube-scheduler.log || true
+        kubectl logs -n kube-system kube-controller-manager-kind-control-plane > $ARTIFACTS/kube-controller-manager.log || true
+        kubectl logs -n kueue-system deployment/kueue-controller-manager > $ARTIFACTS/kueue-controller-manager.log || true
+        kubectl describe pods -n kueue-system > $ARTIFACTS/kueue-system-pods.log || true
```
Comment (ahg-g) [inline hack/e2e-test.sh:14]: jobset instead of kueue :)

Comment (ahg-g) [reply hack/e2e-test.sh:14]: can we also add the api-server?

**Comment by ahg-g** [comment]
> I don't see any meaningful logs from kube-scheduler for example: https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_jobset/214/pull-jobset-test-e2e-main-1-24/1674410094915227648/artifacts/kube-scheduler.log

is it possible that we are not setting a high enough log level for kube-scheduler?

**Comment by kannon92** [comment]
> > I don't see any meaningful logs from kube-scheduler for example: https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_jobset/214/pull-jobset-test-e2e-main-1-24/1674410094915227648/artifacts/kube-scheduler.log
> 
> is it possible that we are not setting a high enough log level for kube-scheduler?

If we want that I think we can expand our Kind config to increase logs.  But if we need it I think that can be done as a followup  separately?

File: hack/e2e-test.sh [hunk 2]
```diff
@@ -4,15 +4,25 @@ export KIND=$PWD/bin/kind
 
 function cleanup {
     if [ $USE_EXISTING_CLUSTER == 'false' ] 
-    then 
-        $KIND delete cluster --name $KIND_CLUSTER_NAME
+    then
+        if [ ! -d "$ARTIFACTS" ]; then
+            mkdir -p "$ARTIFACTS"
+        fi
+        kubectl logs -n kube-system kube-scheduler-kind-control-plane > $ARTIFACTS/kube-scheduler.log || true
```
Comment (danielvegamyhre) [inline hack/e2e-test.sh:11]: Perhaps we can do this in a separate PR but since ARTIFACTS is set as the `$PROJECT_DIR/bin` folder [here](https://github.com/kubernetes-sigs/jobset/blob/main/Makefile#L40) think we should think about storing logs in a separate directory, rather than the bin folder.

Comment (ahg-g) [reply hack/e2e-test.sh:11]: I can access them from the artifacts tab on prow

Comment (danielvegamyhre) [reply hack/e2e-test.sh:11]: ah ok i see
